File: 240402/좌표평면 위의 특정 구역 2/specific-zone-above-the-2d-coordinate-2.java
import java.io.*;
import java.util.*;

public class Main {
    public static int N;
    public static int[] x;
    public static int[] y;
    public static int answer = Integer.MAX_VALUE;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        N = sc.nextInt();
        x = new int[N];
        y = new int[N];

        for (int i = 0; i < N; i++) {
            x[i] = sc.nextInt();
            y[i] = sc.nextInt();
        }

        for (int i = 0; i < N; i++) {
            int minX = Integer.MAX_VALUE, maxX = 1;
            int minY = Integer.MAX_VALUE, maxY = 1;
            for (int j = 0; j < N; j++) {
                if (i == j)
                    continue;
                minX = Math.min(minX, x[j]);
                maxX = Math.max(maxX, x[j]);
                minY = Math.min(minY, y[j]);
                maxY = Math.max(maxY, y[j]);
            }
            int area = (maxX - minX) * (maxY - minY);
            answer = Math.min(answer, area);
        }
        System.out.println(answer);
    }
}

File: 240402/좌표평면 위의 특정 구역 2/specific-zone-above-the-2d-coordinate-2.py
import sys

MAX_NUM = sys.maxsize
ANS = sys.maxsize
N = int(input())
x = []
y = []
for _ in range(N):
    xInput, yInput = tuple(map(int, input().split()))
    x.append(xInput)
    y.append(yInput)

for i in range(N):
    maxX = 1
    minX = MAX_NUM
    maxY = 1
    minY = MAX_NUM
    for j in range(N):
        if (i == j):
            continue
        maxX = max(maxX, x[j])
        minX = min(minX, x[j])
        maxY = max(maxY, y[j])
        minY = min(minY, y[j])
    ANS = min(ANS, ((maxX - minX) * (maxY - minY)))

print(ANS)